src/NUnitTestAdapter/TestConverter.cs

```diff
@@ -142,11 +139,11 @@ private TestCase MakeTestCaseFromXmlNode(XmlNode testNode)
 
             var className = testNode.GetAttribute("classname");
             var methodName = testNode.GetAttribute("methodname");
-            var navData = GetNavigationData(className, methodName);
-            if (NavigationDataIsValid(navData))
+            var navData = _navigationDataProvider.GetNavigationData(className, methodName);
```

Comment:
The code is a lot cleaner now that the getting of navigation data is pulled out of this class. :+1: